From 7539071c94ad36a295700acd4e5e82b9483953b7 Mon Sep 17 00:00:00 2001
From: Mario Carrillo <mario.alfredo.c.arevalo@intel.com>
Date: Thu, 10 Mar 2016 18:23:20 -0600
Subject: [PATCH] fix combined shared library ABI version

this error is from dpdk upstream, dpdk combined
lib doesn't have ABI version, this produces package
errors.

Signed-off-by: Mario Carrillo <mario.alfredo.c.arevalo@intel.com>
---
 mk/rte.sharelib.mk | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/mk/rte.sharelib.mk b/mk/rte.sharelib.mk
index 70c49a8..9ba20e0 100644
--- a/mk/rte.sharelib.mk
+++ b/mk/rte.sharelib.mk
@@ -36,7 +36,8 @@ VPATH += $(SRCDIR)
 
 ifeq ($(CONFIG_RTE_BUILD_COMBINE_LIBS),y)
 ifeq ($(CONFIG_RTE_BUILD_SHARED_LIB),y)
-LIB_ONE := lib$(RTE_LIBNAME).so
+LIB_ONE := lib$(RTE_LIBNAME).so.2.2
+LIB_SONAME := lib$(RTE_LIBNAME).so.2
 else
 LIB_ONE := lib$(RTE_LIBNAME).a
 endif
@@ -54,10 +55,10 @@ ifeq ($(LINK_USING_CC),1)
 LD := $(CC) $(CPU_CFLAGS)
 O_TO_S = $(LD) $(call linkerprefix,$(CPU_LDFLAGS)) \
 	 $(call linkerprefix,$(COMBINED_LDFLAGS)) \
-	-shared $(OBJS) -o $(RTE_OUTPUT)/lib/$(LIB_ONE)
+	-shared -Wl,-soname,$(LIB_SONAME) $(OBJS) -o $(RTE_OUTPUT)/lib/$(LIB_ONE)
 else
 O_TO_S = $(LD) $(CPU_LDFLAGS) $(COMBINED_LDFLAGS) \
-	-shared $(OBJS) -o $(RTE_OUTPUT)/lib/$(LIB_ONE)
+	-shared -soname=$(LIB_SONAME )$(OBJS) -o $(RTE_OUTPUT)/lib/$(LIB_ONE)
 endif
 
 O_TO_S_STR = $(subst ','\'',$(O_TO_S)) #'# fix syntax highlight
@@ -84,6 +85,8 @@ $(LIB_ONE): FORCE
 	@[ -d $(dir $@) ] || mkdir -p $(dir $@)
 	@$(SRCDIR)/scripts/merge-maps.sh > $(COMBINED_MAP)
 	$(O_TO_S_DO)
+	$(Q)ln -s -f $(LIB_ONE) $(basename $(basename $(RTE_OUTPUT)/lib/$(LIB_ONE)))
+	$(Q)ln -s -f $(LIB_ONE) $(basename $(RTE_OUTPUT)/lib/$(LIB_ONE))
 else
 $(LIB_ONE): FORCE
 	@[ -d $(dir $@) ] || mkdir -p $(dir $@)
-- 
2.6.4.442.g545299f.dirty

